core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -504,7 +504,8 @@ private[spark] class BlockManager(
 
     hostLocalDirManager = {
       if (conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED) &&
-          !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)) {
+          !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL) ||
+          Utils.isPushBasedShuffleEnabled(conf)) {
```

Comment:
This still looks the same @zhouyejoe. `hostLocalDirManager` should be set when push-based shuffle is enable. Can you put `()` around  `conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED) &&
          !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)` so that it is clear.
It should be:
```
if (conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED) &&
          !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL) ||
          Utils.isPushBasedShuffleEnabled(conf))
 ```

Reply:
I agree with @otterc's comment - though I think the suggested change in the comment does not do what you want it to do Chandni ? :-)

Reply:
Yeah you are right Mridul. It should be 
```
if ((conf.get(config.SHUFFLE_HOST_LOCAL_DISK_READING_ENABLED) &&
          !conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)) ||
          Utils.isPushBasedShuffleEnabled(conf))
```